org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -0,0 +1,196 @@
+* Contributors:                                                       *
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.window;
+
+import org.adempiere.webui.component.Button;
+import org.adempiere.webui.component.ConfirmPanel;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.Window;
+import org.adempiere.webui.editor.WRecordIDEditor;
+import org.adempiere.webui.editor.WSearchEditor;
+import org.adempiere.webui.editor.WTableDirEditor;
+import org.adempiere.webui.factory.ButtonFactory;
+import org.compiere.model.GridField;
+import org.compiere.model.MColumn;
+import org.compiere.model.MLookup;
+import org.compiere.model.MLookupFactory;
+import org.compiere.model.MLookupInfo;
+import org.compiere.model.MTable;
+import org.compiere.util.DisplayType;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.zkoss.zhtml.Text;
+import org.zkoss.zk.ui.Page;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.EventListener;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+import org.zkoss.zul.Vlayout;
+
+/**
+ * 
+ * @author Peter Takacs, Cloudempiere
+ *
+ */
+public class WRecordIDDialog extends Window implements EventListener<Event> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 8126107952514403099L;
+	
+	private Page page;
+	private WRecordIDEditor editor;
+	private GridField tableIDGridField;
+	
+	private Div contentDiv;
+	private Div labelsDiv;
+	private Div fieldsDiv;
+	private Div confirmPanelDiv;
+	private WTableDirEditor tableIDEditorCopy;
+	
+	private Button okBtn;
+	private Button cancelBtn;
+	private Textbox parentTextBox;
+	private WSearchEditor recordsEditor;
+	
+	private Integer value;
+	
+	public WRecordIDDialog(Page page, WRecordIDEditor editor, GridField tableIDGridField) {
+		super();
+		
+		this.page = page;
+		this.editor = editor;
+		this.tableIDGridField = tableIDGridField;
+		if(editor.getValue() instanceof String)
+			this.value = Integer.valueOf((String)editor.getValue());
+		else
+			this.value = (Integer)editor.getValue();
+		
+		init();
+	}
+	
+	private void init() {
+		Vlayout vLayout = new Vlayout();
+		labelsDiv = new Div();
+		fieldsDiv = new Div();
+		contentDiv = new Div();
+		confirmPanelDiv = new Div();
+    	okBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_OK, true, true);;
+    	cancelBtn = ButtonFactory.createNamedButton(ConfirmPanel.A_CANCEL, true, true);;
+    	tableIDEditorCopy = new WTableDirEditor("AD_Table_ID", false, false, true, tableIDGridField.getLookup(), true);
+		tableIDEditorCopy.setValue(tableIDGridField.getValue());
+		MLookup recordsLookup = getRecordsLookup();
+    	recordsEditor = recordsLookup != null ? new WSearchEditor("Record_ID", false, false, true, recordsLookup) : new WSearchEditor(editor.getGridField());
```

Comment:
A WSearchEditor without lookup ? Don't get what's the possible use case here ...